File: problems/python/removeDuplicates.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        tmp = None
        count = 0
        i = 0
        while i < len(nums):
            # print("i:", i, "nums:", nums, "tmp:", tmp, "nums[i]:", nums[i], "count:", count)
            if tmp == nums[i]:
                count += 1
                if count >= 2:
                    # print("count:", count)
                    nums.pop(i)
                    i -= 1
            else:
                count = 0
            tmp = nums[i]
            i += 1

        return len(nums)


if __name__ == "__main__":
    nums, expected = ([1, 1, 1, 2, 2, 3, 3, 3, 3], [1, 1, 2, 2, 3, 3])
    length = Solution().removeDuplicates(nums)
    print(nums)

File: problems/go/removeDuplicates.go
func removeDuplicates(nums []int) int {
    i := 0
    for j := 1; j < len(nums); j ++ {
        if nums[i] != nums[j] {
            i++
            nums[i] = nums[j]
        }
    }
    return i+1
}


// func delete(s []int, i int) []int {
//     s = append(s[:i], s[i+1:]...)
//     return s
// }


// func removeDuplicates(nums []int) int {
//     if len(nums) == 0 {
//         return 0
//     }
//     var currentN int
//     i := 0
//     for {
//         n := nums[i]
//         if i > 0 && n == currentN {
//             nums = delete(nums, i)
//         } else {
//             currentN = n 
//             i++
//         }

//         if i >= len(nums) {
//             break
//         }
//     }
//     return len(nums)
// }

